File: mobile_app/lib/src/models/offer.dart
class Offer {
  final int id;
  final String title;
  final String description;
  final int discountPercentage;
  final String? startDate;
  final String? endDate;
  final String? imagePath;

  Offer({
    required this.id,
    required this.title,
    required this.description,
    this.discountPercentage = 0,
    this.startDate,
    this.endDate,
    this.imagePath,
  });

  factory Offer.fromJson(Map<String, dynamic> json) {
    return Offer(
      id: int.tryParse(json['id'].toString()) ?? 0,
      title: json['title']?.toString() ?? '',
      description: json['description']?.toString() ?? '',
      discountPercentage:
          int.tryParse(json['discount_percentage'].toString()) ?? 0,
      startDate: json['start_date']?.toString(),
      endDate: json['end_date']?.toString(),
      imagePath: json['image_path'] as String?,
    );
  }
}

File: app/Models/Offer.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

/**
 * @property int $id
 * @property int $pharmacy_id
 * @property string $title
 * @property string|null $description
 * @property float $discount_percentage
 * @property \Illuminate\Support\Carbon|null $start_date
 * @property \Illuminate\Support\Carbon|null $end_date
 * @property string|null $image_path
 * @property \Illuminate\Support\Carbon $created_at
 * @property \Illuminate\Support\Carbon $updated_at
 */
class Offer extends Model
{
    protected $fillable = [
        'pharmacy_id',
        'title',
        'description',
        'discount_percentage',
        'start_date',
        'end_date',
        'image_path',
    ];

    protected $casts = [
        'discount_percentage' => 'integer',
        'start_date' => 'date',
        'end_date' => 'date',
    ];

    /**
     * Get the pharmacy that owns the offer.
     *
     * @return BelongsTo<Pharmacy, Offer>
     */
    public function pharmacy(): BelongsTo
    {
        return $this->belongsTo(Pharmacy::class);
    }
}